polaris.shopify.com/src/components/MobileNav/MobileNav.tsx

```diff
@@ -0,0 +1,247 @@
+import { useState, useEffect, useRef } from "react";
+import Link from "next/link";
+import Image from "next/image";
+
+import { foundationsNavItems } from "../../data/navItems";
+import { className, getComponentNav } from "../../utils/various";
+import { Breakpoints } from "../../types";
+import Button from "../Button";
+import type { NavItem } from "../Nav";
+
+import shopifyLogo from "../../../public/shopify-logo.svg";
+import styles from "./MobileNav.module.scss";
+
+const componentsNavItems = getComponentNav();
+
+const navItems: NavItem[] = [
+  {
+    title: "Getting started",
+    url: "/",
+  },
+  ...foundationsNavItems,
+  {
+    title: "Components",
+    url: "/components",
+    children: componentsNavItems[0].children,
+  },
+  {
+    title: "Tokens",
+    url: "/tokens/colors",
+  },
+  {
+    title: "Icons",
+    url: "/icons",
+  },
+  // ...contributingNavItems,
+];
+
+interface Props {
+  currentPath: string;
+}
+
+function MobileNav({ currentPath }: Props) {
+  const [showMenu, setShowMenu] = useState(false);
+  const menuButtonRef = useRef<HTMLButtonElement>(null);
+  const logoRef = useRef<HTMLAnchorElement>(null);
+  const closeButtonRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    function hideSideNavOnResize() {
+      if (window.innerWidth > Breakpoints.Desktop && showMenu) {
+        setShowMenu(false);
+      }
+    }
+
+    window.addEventListener("resize", hideSideNavOnResize);
+
+    return () => window.removeEventListener("resize", hideSideNavOnResize);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm always suspicious of disabled linting 👀 Why do we not want to include `showMenu` among the dependencies here?

Reply:
Because if we pass `showMenu` in the dependencies we will add/remove the event listener every time that prop changes values

When we pass an empty dependency array to `useEffect`, react will only call that effect when "mounting" the component

This listener only needs to be added once in the first load of a page and never more. It also only covers a very edge case of a user who opens the website in a small browser and resizes it while the mobile nav is opened

Reply:
Great explaination! Thank you!